app/src/main/java/com/lastaoutdoor/lasta/ui/screen/discover/FilterScreen.kt

```diff
@@ -19,190 +21,207 @@ import androidx.compose.material3.ExperimentalMaterial3Api
   }
 
-  Column(
-      modifier = Modifier.fillMaxSize().padding(16.dp).testTag("filterScreen"),
-      horizontalAlignment = Alignment.CenterHorizontally) {
+  var checkedBox by remember { mutableStateOf(true) }
 
-        // return button to go back to the discovery screen
-        TopAppBar(
-            title = { Text(text = stringResource(id = R.string.filter_options)) },
-            navigationIcon = {
-              IconButton(onClick = { navigateBack() }) {
-                Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = null)
-              }
-            })
-
-        Spacer(modifier = Modifier.height(16.dp))
-
-        // Filter by activity type
-        Text(
-            text = stringResource(id = R.string.filter_activity_type),
-            style = TextStyle(fontSize = 20.sp, lineHeight = 24.sp, fontWeight = FontWeight(500)))
-        Row(verticalAlignment = Alignment.CenterVertically) {
-          activities.forEachIndexed { index, activityType ->
-            RadioButton(selected = index == selectedIndex, onClick = { selectedIndex = index })
-            Text(text = activityType.toString())
+  Column(modifier = Modifier.fillMaxSize().testTag("filterScreen")) {
+    MediumTopAppBar(
+        title = {
+          Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Center) {
+            Text(
+                text = stringResource(id = R.string.filter_options),
+                style = MaterialTheme.typography.titleLarge,
+                color = MaterialTheme.colorScheme.onBackground)
           }
-        }
-
-        Spacer(modifier = Modifier.height(4.dp))
-        HorizontalDivider(thickness = 2.dp, color = PrimaryBlue)
-        Spacer(modifier = Modifier.height(16.dp))
-
-        // Filter by difficulty level
-        Text(
-            text = stringResource(id = R.string.filter_difficulty_level),
-            style = TextStyle(fontSize = 20.sp, lineHeight = 24.sp, fontWeight = FontWeight(500)))
-        Row(verticalAlignment = Alignment.CenterVertically) {
-          UserLevel.values().forEach { levelItem ->
-            RadioButton(
-                selected = activitiesLevelArray[selectedIndex] == levelItem,
-                onClick = {
-                  activitiesLevelArray[selectedIndex] = levelItem
-                  setSelectedLevels(
-                      UserActivitiesLevel(
-                          climbingLevel = activitiesLevelArray[0],
-                          hikingLevel = activitiesLevelArray[1],
-                          bikingLevel = activitiesLevelArray[2]))
-                })
-            Text(text = userLevelToDifficultyLevel(levelItem))
+        },
+        navigationIcon = {
+          IconButton(onClick = { navigateBack() }) {
+            Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = null)
+          }
+        })
+
+    HorizontalDivider(modifier = Modifier.fillMaxWidth())
+
+    Spacer(modifier = Modifier.height(16.dp))
```

Comment:
```suggestion
    Spacer(modifier = Modifier.fillMaxHeight(0.05f)
```

With spacer, instead of harcoding a value (In this case it is not a problem since it renders properly on most of devices), you can use properties like Modifier.fillMaxHeight(0.05f) which will take a percentage of the available height (5% here) and therefore will adapt itself with various screen size.

Reply:
Yes, indeed it could be a problem for smaller screen devices, well pointed out !